beard/utils/names.py

```diff
@@ -46,3 +50,77 @@ def normalize_name(name):
 def name_initials(name):
     """Compute the set of initials of a given name."""
     return set([w[0] for w in name.split()])
+
+RE_APOSTROPHES = re.compile('\'+')
+RE_REMOVE_NON_CHARACTERS = re.compile('[^a-zA-Z\',\s]+')
+DROPPED_AFFIXES = {'a', 'ab', 'am', 'ap', 'abu', 'al', 'auf', 'aus', 'bar',
+                   'bath', 'bat', 'ben', 'bet', 'bin', 'bint', 'd', 'da',
+                   'dall', 'dalla', 'das', 'de', 'degli', 'del', 'dell',
+                   'della', 'dem', 'den', 'der', 'di', 'do', 'dos', 'du', 'e',
+                   'el', 'i', 'ibn', 'im', 'jr', 'l', 'la', 'las', 'le',
+                   'los', 'm', 'mac', 'mc', 'mhic', 'mic', 'o', 'ter', 'und',
+                   'v', 'van', 'vom', 'von', 'zu', 'zum', 'zur'}
+
+
+@memoize
+def dm_tokenize_name(name):
```

Comment:
I suggest having a tokenize_name function that could be reused without the Metaphone step. Other steps like handeling apotrophes and prefixes should be parametrized. Additional parametrized step could also be added which is spliting on the caps before lower-casing the token. So that 'MacDonald' example can be treated and could also pass throught eh prefixes step. So the user can deside on this parametrized steps using a scoring metric over her/his own database.